### PR TITLE
Auth integration issue #39

### LINKED ISSUE #39: Integrate Auth login and signup
**Description**
Allow user to login and signup

**Deliverable:**
Allow user to login and signup

### PR DESCRIPTION
**Pull request by akchen1**
User can now login and sign up

### CODE REVIEW

File: app/src/main/java/com/team41/boromi/auth/SignupFragment.java [hunk 1]
```diff
@@ -30,33 +39,72 @@ public SignupFragment() {
...
   @Override
   public View onCreateView(LayoutInflater inflater, ViewGroup container,
       Bundle savedInstanceState) {
     // Inflate the layout for this fragment
-    return inflater.inflate(R.layout.fragment_signup, container, false);
+    View view = inflater.inflate(R.layout.fragment_signup, container, false);
+    activity = (MainActivity) getActivity();
+    spinner = view.findViewById(R.id.signup_loading);
+    spinner.setVisibility(View.GONE);
+    Button createAccountButton = (Button) view.findViewById(R.id.signup_signup);
```
Comment (blchelle) [inline app/src/main/java/com/team41/boromi/auth/SignupFragment.java]: Cast to button is unnecessary

File: app/src/main/java/com/team41/boromi/auth/SignupFragment.java [hunk 2]
```diff
@@ -30,33 +39,72 @@ public SignupFragment() {
...
+    fullNameInput = view.findViewById(R.id.signup_name);
+    emailInput = view.findViewById(R.id.signup_email);
+    userNameInput = view.findViewById(R.id.signup_username);
+    passwordInput = view.findViewById(R.id.signup_password);
+
+    createAccountButton.setOnClickListener(new OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        String fullName = fullNameInput.getText().toString();
```
Comment (blchelle) [inline app/src/main/java/com/team41/boromi/auth/SignupFragment.java:74]: fullName is unused, do you need me to update authController to add the full name as a parameter in makeSignUpRequest?
Honestly, we might not even need to store the names of the users for this app...

**Review by blchelle — COMMENTED** [review]
lgtm

**Comment by blchelle** [comment]
When signing up a user, the username and email get flip-flopped in the db. I don't think this is related to this ticket so I'll fix it in my user backend ticket

File: app/src/main/java/com/team41/boromi/auth/SignupFragment.java (hunk 2 above)
Comment (mingyaang) [reply app/src/main/java/com/team41/boromi/auth/SignupFragment.java:74]: Yeah we figured this was unneeded, if someone could review the user stories to see if this is a needed field then perhaps we can expand on it

**Review by mingyaang — APPROVED** [review]
lgtm, I also saw this demo yesterday night so it's good 👍

File: app/src/main/java/com/team41/boromi/auth/SignupFragment.java (hunk 2 above)
Comment (blchelle) [reply app/src/main/java/com/team41/boromi/auth/SignupFragment.java:74]: I briefly looked at it and the only thing that is mentioned is user "contact information", so it depends on whether we consider someone's name to be contact information

Comment (blchelle) [reply app/src/main/java/com/team41/boromi/auth/SignupFragment.java:74]: Though if including names is considered an extra feature then it doesn't hurt to keep it

File: app/src/main/java/com/team41/boromi/auth/SignupFragment.java (hunk 1 above)
Comment (akchen1) [reply app/src/main/java/com/team41/boromi/auth/SignupFragment.java]: removed

**Pull request merged**